service/DevHome.Service/DevHomeService.cs

```diff
@@ -3,24 +3,94 @@
 
 using System.Diagnostics;
 using System.Runtime.InteropServices;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Windows.Win32.Foundation;
 
 namespace DevHome.Service.Runtime;
 
 [ComVisible(true)]
-public class DevHomeService : IDevHomeService
+public class DevHomeService : IDevHomeService, IDisposable
 {
+    private readonly Process _owner;
+
+    public event MissingFileProcessLaunchFailureHandler? MissingFileProcessLaunchFailure;
+
+    private TraceEventSession? _etwSession;
+
     public DevHomeService()
     {
         Process myCaller = ComHelpers.GetClientProcess();
-
         ComHelpers.VerifyCaller(myCaller);
 
+        _owner = myCaller;
+        _owner.EnableRaisingEvents = true;
+
         // Track our caller process
-        ServiceLifetimeController.RegisterProcess(myCaller);
+        ServiceLifetimeController.ServiceStop += ServiceLifetimeController_ServiceStop;
+        ServiceLifetimeController.RegisterProcess(_owner);
+
+        var crashDumpAnalyzerThread = new Thread(() =>
+        {
+            KernelEventETWListener();
+        });
+        crashDumpAnalyzerThread.Name = "KernelEventETWListenerThread";
+        crashDumpAnalyzerThread.Start();
+
+        _owner.Exited += Owner_Exited;
+    }
+
+    private void ServiceLifetimeController_ServiceStop()
+    {
+        // Be sure to stop our ETW session when we exit. It's possible this gets called multiple times
+        // if we have multiple instances of our object... that's ok.
+        _etwSession?.Stop();
+    }
+
+    private void Owner_Exited(object? sender, EventArgs e)
+    {
+        _etwSession?.Source.StopProcessing();
```

Comment:
This particular call stops processing events from this particular ETW session (but keeps the session open in case there are other listeners).

The stop above closes the ETW session. I'll add some comments.